packages/cli/src/utils/utils/getContractData.ts

```diff
@@ -21,12 +34,40 @@ export function getContractData(
 
   const bytecode = data?.bytecode?.object;
   if (!bytecode) throw new MUDError(`No bytecode found in ${contractDataPath}`);
+  const linkedBytecode = linkLibraries(bytecode, data?.bytecode?.linkReferences, libraries, deployerAddress);
 
   const deployedBytecode = data?.deployedBytecode?.object;
   if (!deployedBytecode) throw new MUDError(`No deployed bytecode found in ${contractDataPath}`);
+  const linkedDeployedBytecode = linkLibraries(
+    deployedBytecode,
+    data?.deployedBytecode?.linkReferences,
+    libraries,
+    deployerAddress,
+  );
 
   const abi = data?.abi;
   if (!abi) throw new MUDError(`No ABI found in ${contractDataPath}`);
 
-  return { abi, bytecode, deployedBytecodeSize: size(deployedBytecode as Hex) };
+  return { abi, bytecode: linkedBytecode, deployedBytecodeSize: size(linkedDeployedBytecode as Hex) };
+}
+
+function linkLibraries(
+  bytecode: Hex,
+  linkReferences: LinkReferences,
+  libraries: PublicLibrary[],
+  deployerAddress: Hex,
+) {
+  let result = bytecode;
+  for (const [filename, referencedLibraries] of Object.entries(linkReferences)) {
+    for (const name of Object.keys(referencedLibraries)) {
+      const fullyQualifiedName = `${filename}:${name}`;
+      const placeholderData = libraries.find((library) => library.fullyQualifiedName === fullyQualifiedName);
+      if (placeholderData === undefined) {
+        throw new Error(`Unlinked library ${fullyQualifiedName}`);
+      }
+      const trimmedAddress = placeholderData.getAddress(deployerAddress).slice(2).toLowerCase();
```

Comment:
To get this address, I had to change when the deterministic deployer is ensured, and pass `deployerAddress` all the through from `runDeploy` => `resolveConfig` => `getPublicLibraries` => `getContractData` => `linkLibraries`

curious if there is a better way